dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/eutaxonomynonfinancials/custom/EuTaxonomyAlignedActivitiesComponent.kt

```diff
@@ -0,0 +1,76 @@
+import org.dataland.frameworktoolbox.intermediate.components.requireDocumentSupportIn
+import org.dataland.frameworktoolbox.intermediate.datapoints.NoDocumentSupport
+import org.dataland.frameworktoolbox.specific.datamodel.TypeReference
+import org.dataland.frameworktoolbox.specific.datamodel.elements.DataClassBuilder
+import org.dataland.frameworktoolbox.specific.fixturegenerator.elements.FixtureSectionBuilder
+import org.dataland.frameworktoolbox.specific.uploadconfig.elements.UploadCategoryBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+import org.dataland.frameworktoolbox.utils.typescript.TypeScriptImport
+
+/**
+ * Represents the EuTaxonomy-Specific "EuTaxonomyAlignedActivities" component
+ */
+class EuTaxonomyAlignedActivitiesComponent(
+    identifier: String,
+    parent: FieldNodeParent,
+) : ComponentBase(
+    identifier, parent,
+) {
+    override fun generateDefaultViewConfig(sectionConfigBuilder: SectionConfigBuilder) {
+        sectionConfigBuilder.addStandardCellWithValueGetterFactory(
+            this,
+            FrameworkDisplayValueLambda(
+                "formatEuTaxonomyNonFinancialsAlignedActivitiesDataForTable(" +
+                    "${getTypescriptFieldAccessor(true)}, \"${
+                        StringEscapeUtils.escapeEcmaScript(
+                            label,
+                        )
+                    }\")",
+                setOf(
+                    TypeScriptImport(
+                        "formatEuTaxonomyNonFinancialsAlignedActivitiesDataForTable",
+                        "@/components/resources/dataTable/conversion/" +
+                            "EuTaxonomyNonFinancialsAlignedActivitiesDataGetterFactory",
+                    ),
+                ),
+            ),
+        )
+    }
+
+    override fun generateDefaultDataModel(dataClassBuilder: DataClassBuilder) {
+        dataClassBuilder.addProperty(
+            this.identifier,
+            TypeReference(
+                "kotlin.collections.MutableList",
+                true,
+                listOf(
+                    TypeReference(
+                        "org.dataland.datalandbackend.frameworks.eutaxonomynonfinancials" +
+                            ".custom.EuTaxonomyAlignedActivity",
+                        false,
+                    ),
+                ),
+            ),
+        )
+    }
+
+    override fun generateDefaultFixtureGenerator(sectionBuilder: FixtureSectionBuilder) {
+        requireDocumentSupportIn(setOf(NoDocumentSupport))
+        sectionBuilder.addAtomicExpression(
+            identifier,
+            "dataGenerator.randomArray(() => dataGenerator.generateAlignedActivity(), 0, 10)",
```

Comment:
Sollte es nicht abhängig von nullable sein ob hier randomArray oder generateArray verwendet wird? Siehe
```
        sectionBuilder.addAtomicExpression(
            identifier,
            documentSupport.getFixtureExpression(
                fixtureExpression = "dataGenerator.guaranteedDecimalYearlyTimeseriesData" +
                    "($jsIdentifierArray, ${uploadBehaviour.yearsIntoPast}, ${uploadBehaviour.yearsIntoFuture})",
                nullableFixtureExpression = "dataGenerator.randomDecimalYearlyTimeseriesData" +
                    "($jsIdentifierArray, ${uploadBehaviour.yearsIntoPast}, ${uploadBehaviour.yearsIntoFuture})",
                nullable = isNullable,
            ),
        )```

Reply:
Ja hast recht. 

Ich glaube am Anfang des tickets hatten wir die custom Sachen alle noch statisch als "immer-nullable" angenommen, weil das in main halt einfach alles nullable ist.

Aber jetzt will man eigentlich, dass non-nullable werden wenn das im data dictionary explizit vorgesehen ist.

Habe es angepasst.